test/javascripts/acceptance/run-query-test.js

```diff
@@ -233,6 +248,74 @@ acceptance("Data Explorer Plugin | Run Query", function (needs) {
     assert.dom("canvas").exists("the chart was rendered");
   });
 
+  test("runs query and is able to download the results", async function (assert) {
+    await visit("/admin/plugins/explorer/queries/-6");
+
+    await click("form.query-run button");
+
+    const createElement = document.createElement.bind(document);
```

Comment:
Personally I would have written this as a system spec to avoid needing to do all these mocks  and stubs, but it's cool you got it to work this way 👍